src/services/atomic-swap.service.js

```diff
@@ -50,10 +52,70 @@ const serviceCreate = async (partialTx, password) => {
     throw new Error('Unable to create the proposal on the Atomic Swap Service');
   }
 
+  await addListenedProposal(walletId, id, password);
   return { proposalId: id };
 };
 
+/**
+ * Adds a new proposal to the `listenedProposals` map of a wallet.
+ * @param {string} walletId
+ * @param {string} proposalId
+ * @param {string} password
+ * @return {Promise<void>}
+ */
+const addListenedProposal = async (walletId, proposalId, password) => {
+  // Checking if this wallet map has been initialized already
+  if (!walletListenedProposals.has(walletId)) {
+    walletListenedProposals.set(walletId, new Map());
+    // Will also open the websocket channel
```

Comment:
Maybe add a TODO? So it's easier to find the places to change later.

If you do it, just remember to change below as well.

Reply:
Added the TODO clauses on [24b3998](https://github.com/HathorNetwork/hathor-wallet-headless/pull/279/commits/24b39981f78fa1d1031448d2434b827e656d4df7)